### DISCUSSION TITLE
Problem with running interactions.s.sol

### DISCUSSION THREAD

**Discussion by owanemi** [Cyfrin/foundry-full-course-cu#2163] [Q&A] [answered]
I GET THIS ERROR WHEN I RUN MY INTERACTIONS.S.SOL
```
script failed: revert: No deployment artifacts were found for specified chain
```
This is the command i used and my terminal output

```
$ forge script script/Interactions.s.sol:FundFundMe --rpc-url $SEPOLIA_URL --private-key $PK  --broadcast
[⠒] Compiling...
No files changed, compilation skipped
Traces:
  [949486] → new FundFundMe@0x5b73C5498c1E3b4dbA84de0F1833c4a029d90519
    └─ ← [Return] 4632 bytes of code

  [294034] FundFundMe::run()
    ├─ [0] VM::readDir("./broadcast", 3) [staticcall]
    │   └─ ← [Return] [("", "C:/Users/Owanemi/Desktop/cyfrin-foundry/foundry-fundme/broadcast\\DeployFundMe.s.sol", 1, true, false), ("", "C:/Users/Owanemi/Desktop/cyfrin-foundry/foundry-fundme/broadcast\\DeployFundMe.s.sol\\11155111", 2, true, false), ("", "C:/Users/Owanemi/Desktop/cyfrin-foundry/foundry-fundme/broadcast\\DeployFundMe.s.sol\\11155111\\run-1722688119.json", 3, false, false), ("", "C:/Users/Owanemi/Desktop/cyfrin-foundry/foundry-fundme/broadcast\\DeployFundMe.s.sol\\11155111\\run-latest.json", 3, false, false), ("", "C:/Users/Owanemi/Desktop/cyfrin-foundry/foundry-fundme/broadcast\\DeployFundMe.s.sol\\31337", 2, true, false), ("", "C:/Users/Owanemi/Desktop/cyfrin-foundry/foundry-fundme/broadcast\\DeployFundMe.s.sol\\31337\\dry-run", 3, true, false)]
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    └─ ← [Revert] revert: No deployment artifacts were found for specified chain


Error:
script failed: revert: No deployment artifacts were found for specified chain
```

And this is my foundry.toml
```
[profile.default]
src = "src"
out = "out"
libs = ["lib"]

fs_permissions = [{ access = "read", path = "./broadcast" }]
# See more config options https://github.com/foundry-rs/foundry/blob/master/crates/config/README.md#all-options
```

**Comment by EngrPips** [comment]
Hello @owanemi, The error said you haven't deployed any `FundFundMe` contract on the Sepolia chain. You should first deploy a `FundFundMe` contract to the Sepolia chain before trying to interact with a `FundFundMe` contract on the Sepolia chain

**Reply by owanemi** [reply]
Isn't the  ```forge script script/Interactions.sol:FundFundMe --rpc-url $SEPOLIA_URL --private-key $PK  --broadcast``` meant to deploy the FundFundMe or i have to do something prior?

**Reply by EngrPips** [reply]
Show your `FundFundMe` script, and let's see if it deploys a `FundMe` contract in its process.

**Reply by owanemi** [reply]
This is the FundFundMe script
```
// SPDX-License-Identifier: MIT
pragma solidity 0.8.19;

import {Script, console} from "lib/forge-std/src/Script.sol";
import {DevOpsTools} from "lib/foundry-devops/src/DevOpsTools.sol";
import {FundMe} from "../src/FundMe.sol";

contract FundFundMe is Script {
    uint256 constant FUND_VALUE = 0.01 ether;

    function fundFundMe(address mostRecentlyDeployed) public {
        vm.startBroadcast();
        FundMe(payable(mostRecentlyDeployed)).fund{value: FUND_VALUE}();
        vm.stopBroadcast();
        console.log("Funded with %s", FUND_VALUE);
    }

    function run() external {
        address mostRecentlyDeployedContract = DevOpsTools.get_most_recent_deployment("FundMe", block.chainid);
        fundFundMe(mostRecentlyDeployedContract);
    }
}

contract WithdrawFundMe is Script {}
```

**Reply by EngrPips** [reply, marked as answer]
> This is the FundFundMe script
> 
> ```
> // SPDX-License-Identifier: MIT
> pragma solidity 0.8.19;
> 
> import {Script, console} from "lib/forge-std/src/Script.sol";
> import {DevOpsTools} from "lib/foundry-devops/src/DevOpsTools.sol";
> import {FundMe} from "../src/FundMe.sol";
> 
> contract FundFundMe is Script {
>     uint256 constant FUND_VALUE = 0.01 ether;
> 
>     function fundFundMe(address mostRecentlyDeployed) public {
>         vm.startBroadcast();
>         FundMe(payable(mostRecentlyDeployed)).fund{value: FUND_VALUE}();
>         vm.stopBroadcast();
>         console.log("Funded with %s", FUND_VALUE);
>     }
> 
>     function run() external {
>         address mostRecentlyDeployedContract = DevOpsTools.get_most_recent_deployment("FundMe", block.chainid);
>         fundFundMe(mostRecentlyDeployedContract);
>     }
> }
> 
> contract WithdrawFundMe is Script {}
> ```

This script does not deploy a `FundMe` contract, but it tries to get a recently deployed `FundMe` contract from your `broadcast` folder. You should deploy a contract first. Are you following Patrick's course, or are you doing this yourself?

**Reply by owanemi** [reply]
yes, you're right it didnt deploy a FundMe contract at first because i wasn't following the course. thank you pips!

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it.
